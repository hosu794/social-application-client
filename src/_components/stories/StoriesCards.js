import React, { lazy, Suspense } from "react";
import { useSelector } from "react-redux";

import { IStory } from "../../_types";
import PropTypes from "prop-types";
const StoryCard = lazy(() => import("./StoryCard"));

function StoriesCards({ stories }) {
  return (
    <Suspense fallback={<div>Loading...</div>}>
      {stories
        ? stories.map((story) => {
            return (
              <StoryCard
                premiumContent={story.premiumContent}
                key={story.id}
                id={story.id}
                username={story.createdBy.username}
                title={story.title}
                userId={story.createdBy.id}
                description={story.description}
                body={story.body}
                data={story.creationDateTime}
              />
            );
          })
        : ""}
    </Suspense>
  );
}

StoriesCards.propTypes = {
  stories: PropTypes.arrayOf(IStory),
};

export default StoriesCards;
