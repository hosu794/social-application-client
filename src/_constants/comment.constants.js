export const commentConstants = {
  GET_ALL_COMMENTS_REQUEST: "GET_ALL_COMMENTS_REQUEST",
  GET_ALL_COMMENTS_SUCCESS: "GET_ALL_COMMENTS_SUCCESS",
  GET_ALL_COMMENTS_FAILURE: "GET_ALL_COMMENTS_FAILURE",
  GET_COMMENT_BY_ID_REQUEST: "GET_COMMENT_BY_ID_REQUEST",
  GET_COMMENT_BY_ID_SUCCESS: "GET_COMMENT_BY_ID_SUCCESS",
  GET_COMMENT_BY_ID_FAILURE: "GET_COMMENT_BY_ID_FAILURE",
  CREATE_COMMENT_REQUEST: "CREATE_COMMENT_REQUEST",
  CREATE_COMMENT_SUCCESS: "CREATE_COMMENT_SUCCESS",
  CREATE_COMMENT_FAILURE: "CREATE_COMMENT_FAILURE",
  GET_COMMENT_BY_CREATED_BY_REQUEST: "GET_COMMENT_BY_CREATED_BY_REQUEST",
  GET_COMMENT_BY_CREATED_BY_SUCCESS: "GET_COMMENT_BY_CREATED_BY_SUCCESS",
  GET_COMMENT_BY_CREATED_BY_FAILURE: "GET_COMMENT_BY_CREATED_BY_FAILURE",
  DELETE_COMMENT_REQUEST: "DELETE_COMMENT_REQUEST",
  DELETE_COMMENT_SUCCESS: "DELETE_COMMENT_SUCCESS",
  DELETE_COMMENT_FAILURE: "DELETE_COMMENT_FAILURE",
  GET_COMMENTS_BY_STORY_ID_REQUEST: "GET_COMMENTS_BY_STORY_ID_REQUEST",
  GET_COMMENTS_BY_STORY_ID_SUCCESS: "GET_COMMENTS_BY_STORY_ID_SUCCESS",
  GET_COMMENTS_BY_STORY_ID_FAILURE: "GET_COMMENTS_BY_STORY_ID_FAILURE",
  GET_COMMENTS_BY_USER_ID_REQUEST: "GET_COMMENTS_BY_USER_ID_REQUEST",
  GET_COMMENTS_BY_USER_ID_SUCCESS: "GET_COMMENTS_BY_USER_ID_SUCCESS",
  GET_COMMENTS_BY_USER_ID_FAILURE: "GET_COMMENTS_BY_USER_ID_FAILURE",
  UPDATE_COMMENT_REQUEST: "UPDATE_COMMENT_REQUEST",
  UPDATE_COMMENT_SUCCESS: "UPDATE_COMMENT_SUCCESS",
  UPDATE_COMMENT_FAILURE: "UPDATE_COMMENT_FAILURE",
};
