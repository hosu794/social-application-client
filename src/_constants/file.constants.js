export const fileConstants = {
  AVATAR_UPLOAD_REQUEST: "AVATAR_UPLOAD_REQUEST",
  AVATAR_UPLOAD_SUCCESS: "AVATAR_UPLOAD_SUCCESS",
  AVATAR_UPLOAD_FAILURE: "AVATAR_UPLOAD_FAILURE",
};
