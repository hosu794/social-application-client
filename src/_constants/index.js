export * from "./alert.constants";
export * from "./auth.constants";
export * from "./story.constants";
export * from "./user.constants";
export * from "./topic.contants";
export * from "./file.constants";
export * from "./comment.constants";
