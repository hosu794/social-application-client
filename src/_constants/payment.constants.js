export const paymentConstants = {
  PAY_PREMIUM_REQUEST: "PAY_PREMIUM_REQUEST",
  PAY_PREMIUM_SUCCESS: "PAY_PREMIUM_SUCCESS",
  PAY_PREMIUM_FAILURE: "PAY_PREMIUM_FAILURE",
};
