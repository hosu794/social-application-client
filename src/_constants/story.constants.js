export const storyConstants = {
  GETPAGED_STORIES_REQUEST: "GETPAGED_STORIES_REQUEST",
  GETPAGED_STORIES_SUCCESS: "GETPAGED_STORIES_SUCCESS",
  GETPAGED_STORIES_FAILURE: "GETPAGED_STORIES_FAILURE",
  GET_STORY_BY_ID_REQUEST: "GET_STORY_BY_ID_REQUEST",
  GET_STORY_BY_ID_SUCCESS: "GET_STORY_BY_ID_SUCCESS",
  GET_STORY_BY_ID_FAILURE: "GET_STORY_BY_ID_FAILURE",
  GETPAGED_STORIES_BY_USERNAME_REQUEST: "GETPAGED_STORIES_BY_USERNAME_REQUEST",
  GETPAGED_STORIES_BY_USERNAME_SUCCESS: "GETPAGED_STORIES_BY_USERNAME_SUCCESS",
  GETPAGED_STORIES_BY_USERNAME_FAILURE: "GETPAGED_STORIES_BY_USERNAME_FAILURE",
  UPDATE_STORY_REQUEST: "UPDATE_STORY_REQUEST",
  UPDATE_STORY_SUCCESS: "UPDATE_STORY_SUCCESS",
  UPDATE_STORY_FAILURE: "UPDATE_STORY_FAILURE",
  CAST_LOVE_REQUEST: "CAST_LOVE_REQUEST",
  CAST_LOVE_SUCCESS: "CAST_LOVE_SUCCESS",
  CAST_LOVE_FAILURE: "CAST_LOVE_FAILURE",
  UNCAST_LOVE_REQUEST: "UNCAST_LOVE_REQUEST",
  UNCAST_LOVE_SUCCESS: "UNCAST_LOVE_SUCCESS",
  UNCAST_LOVE_FAILURE: "UNCAST_LOVE_FAILURE",
  CHANGE_PAGE: "CHANGE_PAGE",
  CREATE_STORY_REQUEST: "CREATE_STORY_REQUEST",
  CREATE_STORY_SUCCESS: "CREATE_STORY_SUCCESS",
  CREATE_STORY_FAILURE: "CREATA_STORY_FAILURE",
  DELETE_STORY_REQUEST: "DELETE_STORY_REQUEST",
  DELETE_STORY_SUCCESS: "DELETE_STORY_SUCCESS",
  DELETE_STORY_FAILURE: "DELETE_STORY_FAILURE",
};
