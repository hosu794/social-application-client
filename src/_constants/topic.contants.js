export const topicContants = {
  GET_TOPICS_REQUEST: "GET_TOPICS_REQUEST",
  GET_TOPICS_SUCCESS: "GET_TOPICS_SUCCESS",
  GET_TOPICS_FAILURE: "GET_TOPICS_FAILURE",
  GET_TOPIC_BY_NAME_REQUEST: "GET_TOPIC_BY_NAME_REQUEST",
  GET_TOPIC_BY_NAME_SUCCESS: "GET_TOPIC_BY_NAME_SUCCESS",
  GET_TOPIC_BY_NAME_FAILURE: "GET_TOPIC_BY_NAME_FAILURE",
};
