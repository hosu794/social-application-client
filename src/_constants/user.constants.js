export const userConstants = {
  GET_CURRENT_USER_REQUEST: "GET_CURRENT_USER_REQUEST",
  GET_CURRENT_USER_SUCCESS: "GET_CURRENT_USER_SUCCESS",
  GET_CURRENT_USER_FAILURE: "GET_CURRENT_USER_FAILURE",
};
