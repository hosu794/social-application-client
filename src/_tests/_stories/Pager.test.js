import React from "react";

import configureMock from "redux-mock-store";
import { Provider } from "react-redux";
import Pager from "../../_components/stories/Pagination";
import { MemoryRouter } from "react-router-dom";
import { mount } from "enzyme";
import StoriesCards from "../../_components/stories/StoriesCards";
import { storyConstants } from "../../_constants";

const mockStore = configureMock([]);

let mockUser = {
  id: 12,
  name: "Dummy Name",
  description: "Description",
};

let store = mockStore({
  stories: {
    content: [
      {
        id: 1,
        title: "Story Title",
        description: "Story description",
        totalLoves: 12,
        body: "<p>Simple Story</p>",
        createdBy: {
          id: 12,
          name: "Joe Doe",
          username: "exampleusername",
          downloadAvatar: "dsdasdasdasdasdadas.png",
        },
        topic: {
          id: 12,
          title: "Topic Example Title",
          description: "Description Example description",
          createdAt: "Dummy date",
          createdBy: {
            id: 12,
            name: "Joe Doe",
            username: "exampleusername",
            downloadAvatar: "dsdasdasdasdasdadas.png",
          },
        },
      },
    ],
    size: 12,
    totalPages: 2,
    loading: false,
    error: false,
    casting: false,
    currentPage: 0,
  },
});

describe("Test for the Pagination Component without user entity", () => {
  let TestingComponent = () => (
    <MemoryRouter>
      <Provider store={store}>
        <Pager />
      </Provider>
    </MemoryRouter>
  );

  let wrapper = mount(<TestingComponent />);

  store.dispatch = jest.fn();

  test("should render component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should component be defined", () => {
    expect(wrapper).toBeDefined();
  });

  test("should StoriesCards exists", () => {
    expect(wrapper.exists(StoriesCards)).toEqual(true);
  });

  test("should Pagination have correct props", () => {
    let element = wrapper.find(".pagination").first().prop("pages");
    expect(element).toEqual(2);
  });

  test("should Pagination Component have correct props ", () => {
    expect(
      wrapper.find(".pagination").first().props().currentPage
    ).toBeDefined();

    expect(wrapper.find(".pagination").first().props().pages).toBeDefined();
  });
});

describe("Tests for the Pagination Component with user entity", () => {
  let TestingComponent = () => (
    <MemoryRouter>
      <Provider store={store}>
        <Pager user={mockUser} />
      </Provider>
    </MemoryRouter>
  );

  let wrapper = mount(<TestingComponent />);

  test("should render component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should component be defined", () => {
    expect(Pager).toBeDefined();
  });

  test("should StoriesCards exists", () => {
    expect(wrapper.exists(StoriesCards)).toEqual(true);
  });

  test("should Pagination have correct props", () => {
    let element = wrapper.find(".pagination").first().prop("pages");
    expect(element).toEqual(2);
  });

  test("should Pagination Component have correct props ", () => {
    expect(
      wrapper.find(".pagination").first().props().currentPage
    ).toBeDefined();

    expect(wrapper.find(".pagination").first().props().pages).toBeDefined();
  });
});
