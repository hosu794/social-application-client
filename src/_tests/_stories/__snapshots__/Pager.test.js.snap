// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the Pagination Component without user entity should render component correctly 1`] = `ReactWrapper {}`;

exports[`Tests for the Pagination Component with user entity should render component correctly 1`] = `ReactWrapper {}`;
